### DISCUSSION TITLE
Criar um script instalador do MAPOS

### DISCUSSION THREAD

**Discussion by SuperPorte** [RamonSilva20/mapos#1552] [Ideas]
Saudações galera.
O que posso fazer para criar um scrips ou um instalador do MAPOS?
A ultima versão está fantástica, pode TRANQUILAMENTE ficar de sisitema fixo para muito MEI.
Estou querendo entrar no catarse para dar uma forcinha. Assim que eu conseguir me livrar do sistema que atualmente pago, vou reverter tudo para o MAPOS.

**Comment by Pr3d4dor** [comment]
@bernarcom Obrigado por usar o mapos, não entendi muito bem a sua sugestão. O mapos já tem um instalador que pode ser usado.

**Comment by SuperPorte** [comment]
Eita!!Não encontrei não.Eu estou baixando o zip lá no releases.Onde fica o instalador?Em 17 de nov de 2021 20:06, Gianluca Bine ***@***.***> escreveu:

@bernarcom Obrigado por usar o mapos, não entendi muito bem a sua sugestão. O mapos já tem um instalador que pode ser usado.

**Comment by Pr3d4dor** [comment]
@bernarcom Esse vídeo ensina a instalar o mapos e mostra o instalador: https://www.youtube.com/watch?v=8joUGLpzUtE

**Reply by SuperPorte** [reply]
Então...
Acabei de ver o vídeo.
Legal, realmente instala. Mas eu quero criar um .bat ou instalador, mesmo que zip que, depois de executado, abra os arquivos para a pessoa configurar envio de emails, lan, etc e tal. Cada aspecto do sistema ser configurado até por que não sabe nada de sistema.
Tipo:
mapos.bat
abrir site xampp
abrir site composer
.
.
.
abrir config.php
.
.
.
100% funcionando

O que acham?
Tem chance de sucesso?

**Reply by Pr3d4dor** [reply]
@bernarcom Eu acho um pouco difícil pra falar a verdade, e não sei se teria tantas melhorias em relação ao atual.

**Reply by SuperPorte** [reply]
É que reparei que existem coisinhas para configurar que não é exatamente claro pra por 100% do sistema para funcionar.
Exemplo é configurar o mail.php ou o alterar o config.php para trabalhar em rede.
Então estou, a passos de formiga, criando um .bat para guiar a instalação desde o download do xampp até por o envio de email automático.
Mas ainda estou apanhando para fazer meu cron acordar e fazer o envio. Nem log ele está gerando =(
Mas quando eu tinver um script minimamente funcionando, posto aqui para avaliação de vocês :)

**Comment by SuperPorte** [comment]
Boa Noite pessoal!!

Segue minha pequeníssima contribuição ao projeto. Um SCRIPT .bat para instalar o MAPOS:
O script baixa todos os arquivos necessários para instalar o MAPOS em uma pasta de faz a instalação nas pastas mais comuns.
PS: não tive sucesso em atualizar o COMPOSER pelo script. Precisei fazer em um terminal paralelo;
PS2: Não sei como programar o envio de e-mail pelo windows.
Espero que gostem :D

`@echo off
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #                 BEM VINDO AO INSTALADOR MAPOS           #
@echo #                                                         #
@echo #     1- baixando XAMPP7.4, COMPOSER e MAPOS4.35.2        #
@echo #                                                         #
@echo #        AGUARDE o fim do DOWNLOAD                        #
@echo ###########################################################
md c:\InstaladorMAPOS\
START /WAIT PowerShell.exe -NoProfile -ExecutionPolicy ByPass -Command "& Invoke-WebRequest https://www.apachefriends.org/xampp-files/7.4.26/xampp-windows-x64-7.4.26-1-VC15-installer.exe -OutFile c:\InstaladorMAPOS\XAMPP74.exe
START /WAIT PowerShell.exe -NoProfile -ExecutionPolicy ByPass -Command "& Invoke-WebRequest https://getcomposer.org/Composer-Setup.exe -OutFile c:\InstaladorMAPOS\COMPOSER.exe
START /WAIT PowerShell.exe -NoProfile -ExecutionPolicy ByPass -Command "& Invoke-WebRequest https://github.com/RamonSilva20/mapos/archive/refs/tags/v4.35.2.zip -OutFile c:\InstaladorMAPOS\MAPOS.zip
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     2- instalando XAMPP;                                #
@echo ###########################################################
start /wait c:\InstaladorMAPOS\COMPOSER.exe
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     3- instalando COMPOSER;                             #
@echo ###########################################################
start /wait c:\InstaladorMAPOS\COMPOSER.exe
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     4- Instalando MAPOS;                                #
@echo ###########################################################
## PowerShell.exe -NoProfile -ExecutionPolicy ByPass -Command "& Expand-Archive c:\InstaladorMAPOS\MAPOS.zip -DestinationPath c:\xampp\htdocs\
ren c:\xampp\htdocs\mapos-4.35.2 mapos
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     5- Area reservada para atualizar COMPOSER          #
@echo #                                                         #
@echo #     Nao consegui fazer                                  #
@echo #                                                         #
@echo #                                                         #
@echo ###########################################################
set composer="composer install --no-dev"
cd c:\xampp\htdocs\mapos
start /wait %composer%
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     6- Crie o banco de dados MAPOS                      #
@echo ###########################################################
"C:\Program Files\Mozilla Firefox\firefox.exe" http:\\localhost\phpmyadmin
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     7- configurando MAPOS                               #
@echo ###########################################################
"C:\Program Files\Mozilla Firefox\firefox.exe" http:\\localhost\mapos
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     8- Quer o MAPOS acessivel em rede?                  #
@echo #                                                         #
@echo #     S para SIM                                          #
@echo #     N para NAO                                          #
@echo ###########################################################
SET /P rede={S,N}
goto %rede%
:s
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     9- Mude, a linha 42 que esta:                       #
@echo #                                                         #
@echo #     $config['base_url'] = 'enter_baseurl'               #
@echo #     para:                                               #
@echo #     $config['base_url'] = 'http://192.168.1.145/mapos'; #
@echo #                                                         #
@echo ###########################################################
notepad C:\xampp\htdocs\mapos\application\config\config.php
:n
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     10- Quer alterar o numero inicial de suas OSs?      #
@echo #                                                         #
@echo #     1 para SIM                                          #
@echo #     2 para NAO                                          #
@echo ###########################################################
SET /P OSs={1,2}
goto %OSs%
:1
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     11- Configurando Numero inicial OSs                 #
@echo #                                                         #
@echo #     Va no campo AUTO_INCREMENT e mude para o            #
@echo #     conteudo para o numero desejado                     #
@echo #                                                         #
@echo ###########################################################
"C:\Program Files\Mozilla Firefox\firefox.exe" http://localhost/phpmyadmin/index.php?route=/table/operations&db=mapos&table=os
:2
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #     12- Area reservada para configurar email            #
@echo #                                                         #
@echo #     Nao consegui fazer no windows                       #
@echo #                                                         #
@echo #                                                         #
@echo ###########################################################
"C:\Program Files\Mozilla Firefox\firefox.exe" http://localhost/phpmyadmin/index.php?route=/table/operations&db=mapos&table=os
cls
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ..
@echo ###########################################################
@echo #                                                         #
@echo #                                                         #
@echo #      MAPOS configurado e pronto para uso!!              #
@echo #                                                         #
@echo #                                                         #
@echo ###########################################################
`

**Comment by SuperPorte** [comment]
Percebi que o forum "deformou" um pouco o script.
Vou postar o .bat aqui.

O forum não deixou subir o .bat. Vou postar como formato .MD
[InstaladorMAPOS.MD](https://github.com/RamonSilva20/mapos/files/7741859/InstaladorMAPOS.MD)

**Comment by Pr3d4dor** [comment]
@bernarcom Se vocẽ quiser, você pode fazer uma PR com esse script para adicionar ele no repositório oficial.

**Reply by SuperPorte** [reply]
Precisa não hehehe.
É uma contribuiçãozinha minha :D

**Comment by SuperPorte** [comment]
Nova versão do script:
[InstaladorMAPOS ver_1.1.MD](https://github.com/RamonSilva20/mapos/files/7781613/InstaladorMAPOS.ver_1.1.MD)

**Comment by SuperPorte** [comment]
Saudações Galera!!!
CONSEGUI concluir um script .bat pra QUALQUER um conseguir instalar o MAP-OS!!!
[MAPOS-Instalador-2.2.md](https://github.com/RamonSilva20/mapos/files/8309727/MAPOS-Instalador-2.2.md)

TEM que mudar a extensão .md para .bat!!!!

Só não sei ainda acertar o envio de e-mails no windows, mas o script deixa o sistema funcionando!!!!

**Reply by willph** [reply]
> Saudações Galera!!! CONSEGUI concluir um script .bat pra QUALQUER um conseguir instalar o MAP-OS!!! [MAPOS-Instalador-2.2.md](https://github.com/RamonSilva20/mapos/files/8309727/MAPOS-Instalador-2.2.md)
> 
> TEM que mudar a extensão .md para .bat!!!!
> 
> Só não sei ainda acertar o envio de e-mails no windows, mas o script deixa o sistema funcionando!!!!

o envio de email do windows tem que configurar primeiro o smtp no arquivo email.php em application/config e depois cria um tarefa no agendado de tarefa que esta na PR de Luiz.

https://github.com/luizrn/mapos/blob/Cronjobs-no-Windows-para-enviar-os-e-emails/Cronjobs%20no%20Windows%20para%20enviar%20os%20e-emails.md

**Reply by SuperPorte** [reply]
Opa!!! vou ver esse link mais tarde.
Por enquanto, fiquem com mais uma atualização do instalador
[MAPOS-Instalador-3.2.md](https://github.com/RamonSilva20/mapos/files/8311505/MAPOS-Instalador-3.2.md)
:

Não esqueçam de mudar a extensão de .md para .bat

**Reply by SuperPorte** [reply]
quase consegui colocar um alinha no .bat que programa os envios:
chtasks /create /ru administrator /rp admin /tn mapos /sc minute /mo 5 /tr "C:\xampp\php\php.exe C:\xampp\php\php.exe index.php email/process"

só que "index.php email/process" não vai para o campo "iniciar em (opcional)". Ele está ficando dentro do campo de cima "Adicione argumentos (opcional)"

Alguém sabe como faço para definir esse ultimo argumento?

**Reply by willph** [reply]
Irei analisar e te digo

**Comment by willph** [comment]
"iniciar em (opcional)" fica o `C:\xampp\htdocs\mapos` 
em "Adicione argumentos" fica o `index.php email/process`

**Reply by willph** [reply]
Programa/Script fica o `C:\xampp\php\php.exe`

**Reply by SuperPorte** [reply]
Sim, e como informa isso no cmd?

**Comment by SuperPorte** [comment]
Instalador MAP-OS 4.0 saido do forno agora!!!!
[MAPOS-Instalador-4.0.md](https://github.com/RamonSilva20/mapos/files/8312141/MAPOS-Instalador-4.0.md)
Não esqueçam de alterar a extensão .md para .bat!!!

**Comment by willph** [comment]
So usar `%username%` que já trás o nome do usuário precisando digitar apenas a senha.

```
set /p senha=(Digite o senha do usuario administrador e de enter)
rem Schtasks /create /ru %username% /rp %senha% /tn mapos-mail /XML mapos-mail.xml
Schtasks /create /ru %username% /rp %senha% /tn mapos-mail /sc minute /mo 5 /tr "C:\xampp\php\php.exe index.php email/process"
Schtasks /create /ru %username% /rp %senha% /tn mapos-retry /sc minute /mo 5 /tr "C:\xampp\php\php.exe index.php email/process"
```

**Comment by SuperPorte** [comment]
E se o usuário logado não for o administrador?Em 21 de mar de 2022 08:39, Wilmerson ***@***.***> escreveu:

So usar %username% que já trás o nome do usuário precisando digitar apenas a senha.

set /p senha=(Digite o senha do usuario administrador e de enter)

rem Schtasks /create /ru %username% /rp %senha% /tn mapos-mail /XML mapos-mail.xml

Schtasks /create /ru %username% /rp %senha% /tn mapos-mail /sc minute /mo 5 /tr "C:\xampp\php\php.exe index.php email/process"

Schtasks /create /ru %username% /rp %senha% /tn mapos-retry /sc minute /mo 5 /tr "C:\xampp\php\php.exe index.php email/process"

**Comment by SuperPorte** [comment]
Última versão do BAT instalador:

[MAPOS-Instalador-4.3.md](https://github.com/RamonSilva20/mapos/files/8824785/MAPOS-Instalador-4.3.md)

Não esqueça de mudar a extensão do arquivo de .md para .bat

**Comment by SuperPorte** [comment]
Segue última versão do instalador:

[Instalador-mapos-master - Copia.md](https://github.com/RamonSilva20/mapos/files/9821427/Instalador-mapos-master.-.Copia.md)

Como o github não deixa mandar .bat, alterei a extensão do arquivo para .md
ALTERE de .md para .bat para usar o script.
Execute como ADM para funcionar sem surpresas

**Comment by SuperPorte** [comment]
Saudações galera!!
Versão mais nova do meu script instalador do mapos MASTER!!!

Não esqueça de mudar a extensão de .md para .bat
[Instalador-MAPOS-MASTER.md](https://github.com/RamonSilva20/mapos/files/10961947/Instalador-MAPOS-MASTER.md)
